### ISSUE TITLE
\' is not how you escape single quotes in yaml.

### ISSUE THREAD

**Issue by shawncplus** [dotmaster/toYaml#1]
Current lib escapes single quotes by doing \' when you actually escape them by doing ''. http://yaml.org/spec/current.html#id2534365

**Comment by paleite** [comment]
Life saver, right here :) Thank you! :+1:

**Comment by odiseoteam** [comment]
Another Life Here :')

**Comment by SoulReturns** [comment]
Thank you!

**Comment by ghost** [comment]
Thanks for saving my life. :)

**Comment by marcusstenbeck** [comment]
@dotmaster this looks like it's resolved. I suggest closing this.

**Comment by tobilarscheid** [comment]
Worked for me! Thanks!

**Comment by ValerieAnne563** [comment]
THANK YOU!

**Comment by Sequoia** [comment]
wow that is a crazy way to escape quotes!! Never would have guessed in a million years.

**Comment by rudeboybert** [comment]
I tried `\'`, `\\'`, `\\\'`, to no avail. Thank you @Sequoia!!

**Comment by NinoFoxx** [comment]
OMG.. you saved my life.. I've been working with Minecraft plugin configuration since ages and this is the first time I stomp on this issue. Thank you a lot!!

**Comment by CloudSurgeon** [comment]
another life!

**Comment by hckhanh** [comment]
You save another life @shawncplus

**Comment by jjbalcarcel** [comment]
Anyone know how to escape double quotes?

**Comment by dhartsmith** [comment]
> Anyone know how to escape double quotes?

@jjbalcarcel - I believe you prefix them with a \

**Comment by GiovaniFerraroTrivelli** [comment]
Thank you ^^

**Comment by HackerTheMonkey** [comment]
yet another life saved here, thank you

**Comment by ggedde** [comment]
I haven't tried this in a yaml file yet, but it does not work in the command line for adding --extra-vars using a json string.  I have tried \' and '' and '''' and many more. Using a backslash just causes a parse error and using two single quotes '' or four '''' runs fine but when it gets to the server it is just blank.   Because I am using json I need to wrap the --extra-vars with single quotes as the json requires the double quotes. But then I can't get single quotes to work inside a json variables.
ex. 
`--extra-vars '{"foor": "Bar's"}'`
Any thoughts?

**Comment by emper0r** [comment]
yes... another life saved!

**Comment by riddhij16** [comment]
@ggedde did u find a way?

**Comment by ZaphodB** [comment]
<3

**Comment by ggedde** [comment]
@emper0r No, for now I am just removing single quotes from all my variables. Not ideal, but just a quick solution for now.
